src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts

```diff
@@ -228,6 +228,78 @@ export class InstructorCoursesPageComponent implements OnInit {
     });
   }
 
+  /**
+   * Changes the status of an archived course in front-end only.
+   * This is to reduce the need to refresh the entire list of courses multiple times.
+   */
+  changeFrontEndArchiveStatus(courseId: string, toArchive: boolean): void {
```

Comment:
How about combining `ActiveCourseModel` `ArchivedCourseModel` `SoftDeletedCourseModel` into one `CourseModel`?

Reply:
Noted with thanks! I will consider about that.